File: modules/wem-api/src/main/java/com/enonic/wem/api/schema/mixin/MixinName.java
package com.enonic.wem.api.schema.mixin;


import org.apache.commons.lang.StringUtils;

import com.enonic.wem.api.module.ModuleKey;
import com.enonic.wem.api.schema.ModuleBasedName;

public final class MixinName
    extends ModuleBasedName
{
    private MixinName( final ModuleKey moduleKey, final String localName )
    {
        super( moduleKey, localName );
    }

    public static MixinName from( final ModuleKey moduleKey, final String localName )
    {
        return new MixinName( moduleKey, localName );
    }

    public static MixinName from( final String mixinName )
    {
        final String moduleKey = StringUtils.substringBefore( mixinName, SEPARATOR );
        final String localName = StringUtils.substringAfter( mixinName, SEPARATOR );
        return new MixinName( ModuleKey.from( moduleKey ), localName );
    }
}

File: modules/wem-admin-ui/src/main/resources/web/admin/common/js/schema/mixin/MixinName.ts
module api.schema.mixin{

    import ModuleKey = api.module.ModuleKey;

    export class MixinName extends api.module.ModuleBasedName {

        constructor(name:string) {
            api.util.assertNotNull(name, "Mixin name can't be null");
            var parts = name.split(api.module.ModuleBasedName.SEPARATOR);
            super(ModuleKey.fromString(parts[0]), parts[1]);
        }

        equals(o: api.Equitable): boolean {
            if (!api.ObjectHelper.iFrameSafeInstanceOf(o, MixinName)) {
                return false;
            }

            return super.equals(o);
        }
    }
}